### PR TITLE
chore(ci): add linter for GitHub Actions as pre-commit hook

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #1332

### Changes

> Please provide a summary of what's being changed

This introduces [Action Lint](https://github.com/rhysd/actionlint) as a pre-commit hook to catch unforeseen syntax errors to prevent future failures.

It also includes fixes for all issues caught: key but optional fields missing, dangling debug step, and incorrect syntax for new changelog adhoc workflow

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [ ] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [ ] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented
* [ ] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,3 +39,8 @@ repos:
     hooks:
       - id: cfn-python-lint
         files: examples/.*\.(yaml|yml)$
+  - repo: https://github.com/rhysd/actionlint
```
Comment (leandrodamascena) [inline .pre-commit-config.yaml:42]: I think I need help understanding how to use this. Do I need to install actionlint before performing a commit?

1 - `gh pr checkout 1479`
2 - `pre-commit install`
3 - I modified the `auto-merge.yml` file and added a wrong key
4 - `git commit -S -m "chore(test): test lint"`
5 - The result was:

❯ git commit -S -m "chore(maintainer): add Leandro as maintainer"
Check for merge conflicts................................................Passed
Trim Trailing Whitespace.................................................Passed
Check Toml...........................................(no files to check)Skipped
formatting::black....................................(no files to check)Skipped
formatting::isort....................................(no files to check)Skipped
linting::flake8......................................(no files to check)Skipped
markdownlint.........................................(no files to check)Skipped
AWS CloudFormation Linter............................(no files to check)Skipped
```
Lint GitHub Actions workflow files.......................................Failed
- hook id: actionlint
- exit code: 1

Executable `actionlint` not found
```

Comment (heitorlessa) [reply .pre-commit-config.yaml:42]: hmmm investigating..!

Comment (heitorlessa) [reply .pre-commit-config.yaml:42]: Fixed it. It's the first time we encounter a pre-commit hook that tries to use a system binary by default - great catch!

Try again plz

Comment (leandrodamascena) [reply .pre-commit-config.yaml:42]: Working :dancers:.. Good job!
image

**Pull request merged**